saleor/graphql/giftcard/types.py

```diff
@@ -47,18 +55,30 @@ def resolve_display_code(root: models.GiftCard, *_args, **_kwargs):
     @traced_resolver
     def resolve_user(root: models.GiftCard, info):
         requestor = get_user_or_app_from_context(info.context)
-        if requestor_has_access(requestor, root.user, AccountPermissions.MANAGE_USERS):
-            return root.user
+        if requestor_has_access(
+            requestor, root.created_by, AccountPermissions.MANAGE_USERS
+        ):
+            return root.created_by
         raise PermissionDenied()
 
     @staticmethod
     @traced_resolver
     def resolve_code(root: models.GiftCard, info, **_kwargs):
         user = info.context.user
         # Staff user has access to show gift card code only for gift card without user.
-        if user.has_perm(GiftcardPermissions.MANAGE_GIFT_CARD) and not root.user:
+        if user.has_perm(GiftcardPermissions.MANAGE_GIFT_CARD) and not root.created_by:
             return root.code
         # Only user associated with a gift card can see gift card code.
-        if user == root.user:
+        if user == root.created_by:
             return root.code
         return None
+
+    @staticmethod
+    @traced_resolver
```

Comment:
`traced_resolver` should only be used in more complex resolvers or when fetching relations, should be dropped here